public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -315,6 +316,11 @@ export class PanelModel implements DataConfigSource {
     this.events.publish(new RefreshEvent());
   }
 
+  //this is needed when layout is updated programatically (i.e using the inspectJsonTab)
+  updateGridPositionLayout() {
+    this.events.publish(new PanelGridPositionLayoutEvent({ gridPos: this.gridPos, panelId: this.id }));
```

Comment:
This is something that burned me last week, when in panelEdit `this.id` is a "random" number. Maybe use this instead:
```suggestion
    this.events.publish(new PanelGridPositionLayoutEvent({ gridPos: this.gridPos, panelId: this.editSourceId ?? this.id }));
```

Reply:
🙌🏾 thanks a lot, I will take a look